felix/dataplane/linux/bpf_route_mgr.go

```diff
@@ -645,6 +655,44 @@ func (m *bpfRouteManager) onWorkloadEndpointRemove(update *proto.WorkloadEndpoin
 	m.removeWEP(update.Id)
 }
 
+func (m *bpfRouteManager) onBGPConfigUpdate(update *proto.GlobalBGPConfigUpdate) {
+	blockedCIDRs := []string{}
+	blockedCIDRs = append(blockedCIDRs, update.GetServiceClusterCidrs()...)
+	blockedCIDRs = append(blockedCIDRs, update.GetServiceExternalCidrs()...)
+	blockedCIDRs = append(blockedCIDRs, update.GetServiceLoadbalancerCidrs()...)
+
+	cidrsToDel := set.New[ip.CIDR]()
+	cidrsToDel.AddSet(m.blockedCIDRs)
+	m.blockedCIDRs.Clear()
+
+	for _, cidrStr := range blockedCIDRs {
+		cidr, err := ip.ParseCIDROrIP(cidrStr)
+		if err != nil {
+			log.WithError(err).WithField("cidr", cidr).Error(
+				"Failed to parse cidr.")
+		}
+		if uint8(m.ipFamily) != cidr.Version() {
+			continue
+		}
+
+		if _, ok := m.cidrToRoute[cidr]; ok {
```

Comment:
what if it was previously a cidr for a different route? Don't you need to make sure it is a black hole? doesn't it also stay in the cidrsToDel so it get removed later on?